### PR TITLE
Switch to `EmbeddedRiscvJtag` and `openocd-risv`

### PR DESCRIPTION
**Pull request by lmbollen**
We previously used the old `DebugPlugin` for `VexRiscv` which did not allow us to verify the contents of the loaded elf file on the cpu with `compare-sections`.

This pull requests does the following this:
- Bump scala and spinal versions.
- Use `EmbeddedRiscvJtag` instead of `DebugPlugin`.
- Switch to using `openocd` with `remote-bitbang` as adapter instead of `openocd-vexriscv`s custom `jtag-tcp`.

In the Nix configuration we now use `openocd-riscv` instead of `openocd-vexriscv`, so we should update the docker image.

### CODE REVIEW

File: clash-vexriscv/src/ffi/impl.cpp [hunk 1]
```diff
@@ -229,78 +230,110 @@ vexr_jtag_bridge_data *vexr_jtag_bridge_init(uint16_t port)
...
+								}
+								case 's': { // Reset 0 1
+									break;
+								}
+								case 't': { // Reset 1 1
+									break;
+								}
+								case 'u': { // Reset 1 0
+									break;
+								}
```
Comment (martijnbastiaan) [inline clash-vexriscv/src/ffi/impl.cpp:326]: The heck happened here :see_no_evil:

File: clash-vexriscv-sim/data/vexriscv_chain_sim.cfg [hunk 2]
```diff
@@ -9,32 +9,27 @@
 
 # See vexriscv_sim.cfg for more information on each step taken.
 
-adapter driver jtag_tcp
-adapter speed 64000
+adapter driver remote_bitbang
```
Comment (martijnbastiaan) [inline clash-vexriscv-sim/data/vexriscv_chain_sim.cfg:12]: I get that we switched drivers, but could you elaborate on _why_ we switched drivers? In the commit body  of "[Switch to openocd-vexriscv and use remote-bitbang](https://github.com/clash-lang/clash-vexriscv/pull/45/commits/556b479091a8d8ac9febb709417f40308b67c86d)" please.

File: .github/docker/Dockerfile [hunk 3]
```diff
@@ -18,7 +18,7 @@ RUN apt-get update \
   && apt-get clean \
   && rm -rf /var/lib/apt/lists/*
 
-FROM builder AS build-openocd-vexriscv
+FROM builder AS build-openocd-riscv
```
Comment (martijnbastiaan) [inline .github/docker/Dockerfile:21]: I'll see if I can upstream `openocd-riscv` to `nixpkgs`.

**Review by martijnbastiaan — APPROVED** [review]
Wohoo, LGTM.

Would you mind cherry-picking https://github.com/clash-lang/clash-vexriscv/commit/6909b61b2765e8197009fcbf9d307dd35f9fdaf5?

File: .github/docker/Dockerfile (hunk 3 above)
Comment (martijnbastiaan) [reply .github/docker/Dockerfile:21]: Just to be clear: no action needed for this PR.

File: clash-vexriscv/src/ffi/impl.cpp (hunk 1 above)
Comment (lmbollen) [reply clash-vexriscv/src/ffi/impl.cpp:326]: These are [remote-bitbang](https://github.com/openocd-org/openocd/blob/master/doc/manual/jtag/drivers/remote_bitbang.txt#L51-L73) commands. There's not really a way for us to blink other than maybe using `printf` .For the resets, I have not seen them being driven in simulation and don't know how we'd have to wire them up in simulation.

File: clash-vexriscv-sim/data/vexriscv_chain_sim.cfg (hunk 2 above)
Comment (lmbollen) [reply clash-vexriscv-sim/data/vexriscv_chain_sim.cfg:12]: `jtag_tcp` is not a normal openocd adapter, it was custom added to `openocd-vexriscv` and is very similar to the existing `remote-bitbang` target. I'll add it to the commit message

File: clash-vexriscv/src/ffi/impl.cpp (hunk 1 above)
Comment (martijnbastiaan) [reply clash-vexriscv/src/ffi/impl.cpp]: I meant the weird indentation. It suddenly jumps by 40 characters or so.

**Pull request merged**